frontend/src/RestaurantShow/ReviewForm.js

```diff
@@ -0,0 +1,138 @@
+/* eslint-disable camelcase */
+import React, { useCallback, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { useParams } from "react-router";
+import styled from "styled-components";
+
+import ReactStars from "react-rating-stars-component";
+import { Body, Close, Modal } from "@zendeskgarden/react-modals";
+import { createReview } from "../actions/reviewActions";
+import { selectCurrentUser } from "../reducers/selectors";
+
+import SubmitButton from "../components/SubmitButton";
+
+const ReviewButton = styled.button`
+  border-radius: 5px;
+  border: 1px solid #f40d15;
+  width: 200px;
+  margin: 10px;
+  padding: 10px;
+  font-weight: bold;
+  background-color: #f40d15;
+  font-size: 18px;
+  color: white;
+  height: 20%;
+  outline: none;
+  box-shadow: none;
+
+  &:hover {
+    background-color: #d10f15;
+    transition: border-color 0.25s ease-in-out 0s,
+      box-shadow 0.1s ease-in-out 0s, background-color 0.25s ease-in-out 0s,
+      color 0.25s ease-in-out 0s;
+  }
+
+  &:active {
+    background-color: #8f2821;
+    transition: border-color 0.25s ease-in-out 0s,
+      box-shadow 0.1s ease-in-out 0s, background-color 0.25s ease-in-out 0s,
+      color 0.25s ease-in-out 0s;
+  }
+
+  &:focus {
+    box-shadow: #eb7b73 0px 0px 0px 3px;
+    transition: border-color 0.25s ease-in-out 0s,
+      box-shadow 0.1s ease-in-out 0s, background-color 0.25s ease-in-out 0s,
+      color 0.25s ease-in-out 0s;
+  }
+`;
+const StyledButtonDiv = styled.div`
+  text-align: center;
+`;
+
+const StyledForm = styled.form`
+  display: flex;
+  flex-direction: column;
+  text-align: center;
+  justify-content: center;
+  align-items: center;
+`;
+
+const ReviewForm = () => {
+  const [body, setBody] = useState("");
+  const [rating, setRating] = useState(0);
+  const [visible, setVisible] = useState(false);
+  const dispatch = useDispatch();
+  const currentUser = useSelector(selectCurrentUser);
+  const user_id = currentUser.id;
+  const queryParams = useParams();
+  const restaurant_id = parseInt(queryParams.id, 10);
+
+  const processForm = useCallback((review) => dispatch(createReview(review)), [
+    dispatch,
+  ]);
+
+  const updateBody = useCallback((event) => {
+    setBody(event.currentTarget.value);
+  }, []);
+
+  const updateRating = useCallback((event) => {
+    setRating(event);
+  }, []);
+
+  const handleSubmit = useCallback(
+    (event) => {
+      event.preventDefault();
+      const review = { body, rating, user_id, restaurant_id };
+
+      processForm(review).then(() => {
+        setVisible(false);
```

Comment:
Uh oh, memory leak. In the case that the promise is kicked off and then the component is no longer rendered, what will happen? You need some kind of isMountedRef that gets set to false on unmount about this.